### PR TITLE
Feat/prop types

### PR DESCRIPTION
**Pull request by daltobello**
### Summary:
- File(s) added/changed: App.js
- Description: Added PropType validation where props are passed. Also added network request error handling within App to display a message to the user indicating something went wrong.

### Type of change

- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Comment:

### Checklist:

- [x]  Code needs to be tested
- [ ]  All tests are passing

### CODE REVIEW

**Review by loganpaulmatheny — COMMENTED** [review]
Coded together - looks great!

File: src/App/App.js [hunk 1]
```diff
@@ -19,18 +20,19 @@ function App() {
   const getMovies = () => {
     getAllMovies().then((movieData) => {
       setMovies(movieData.movies);
-    });
+    })
+    .catch(() => setError("Oops...looks like R A N C I D tomatillos rotted..."));
```
Comment (hfaerber) [inline src/App/App.js:24]: Good use of your .catch here

File: src/App/App.js [hunk 2]
```diff
@@ -19,18 +20,19 @@ function App() {
...
   };
 
   const viewMovieBlowUp = (id) => {
```
Comment (hfaerber) [inline src/App/App.js:27]: I like that you've declared this function in App then are passing it as a prop to be invoked elsewhere.  I'm curious if this approach will need to change once you implement router.

File: src/Components/AllMovies/AllMovies.js [hunk 3]
```diff
@@ -1,24 +1,26 @@
-import './AllMovies.css'
-import MovieCard from '../MovieCards/MovieCards'
+import "./AllMovies.css";
+import MovieCard from "../MovieCards/MovieCards";
+import PropTypes from "prop-types";
 
-function AllMovies({moviesData, viewMovieBlowUp}) {
+function AllMovies({ moviesData, viewMovieBlowUp }) {
```
Comment (hfaerber) [inline src/Components/AllMovies/AllMovies.js:5]: Good destructuring here!

File: src/Components/AllMovies/AllMovies.js [hunk 4]
```diff
@@ -1,24 +1,26 @@
...
-  )
+    );
+  });
+  return <div className="movie-container">{movieCards}</div>;
 }
 
-export default AllMovies
\ No newline at end of file
+AllMovies.propTypes = {
+  moviesData: PropTypes.array.isRequired,
```
Comment (hfaerber) [inline src/Components/AllMovies/AllMovies.js:22]: This is a good start on PropTypes but doesn't dig quite deep enough.  Can you go further to specify and enforce that the contents of the array are what you expect?  Should it be an array of strings? Numbers? Objects?

File: src/Components/MovieBlowUp/MovieBlowUp.js [hunk 5]
```diff
@@ -1,10 +1,15 @@
...
+function MovieBlowUp({ movieBlowUp, viewHome }) {
   return (
-    <div >
+    <div>
       <h2>{movieBlowUp.title}</h2>
-      <img className='movie-blowup-poster' src={movieBlowUp.poster_path} alt={`${movieBlowUp.title} poster`}/>
+      <img
+        className="movie-blowup-poster"
+        src={movieBlowUp.poster_path}
+        alt={`${movieBlowUp.title} poster`}
```
Comment (hfaerber) [inline src/Components/MovieBlowUp/MovieBlowUp.js:11]: Great job including the alt tag!

File: src/Components/MovieBlowUp/MovieBlowUp.js [hunk 6]
```diff
@@ -13,9 +18,17 @@ function MovieBlowUp({movieBlowUp, viewHome}) {
...
       <button onClick={() => viewHome()}>Return to all movies</button>
     </div>
-  )
+  );
 }
 
-export default MovieBlowUp 
-
+MovieBlowUp.propTypes = {
+  movieBlowUp: PropTypes.shape({
```
Comment (hfaerber) [inline src/Components/MovieBlowUp/MovieBlowUp.js:25]: Love seeing the use of PropTypes.shape() here!  Nice work!

**Review by hfaerber — COMMENTED** [review]
Great start y'all!  I've left some comments for your review.

**Pull request merged**